Source/websocket/WebSocketLink.h

```diff
@@ -1022,9 +1022,41 @@ POP_WARNING()
         {
             return (_channel.AbortUpgrade(status, reason));
         }
+        uint32_t WaitForLink(const uint32_t time) const
+        {
+            const int SLEEPSLOT_TIME = 100;
+            // Make sure the state does not change in the mean time.
+            Lock();
+
+            uint32_t waiting = (time == Core::infinite ? Core::infinite : time); // Expect time in MS.
+
+            // Right, a wait till connection is closed is requested..
+            while ((waiting > 0) && (IsWebSocket() == false)) {
+                uint32_t sleepSlot = (waiting > SLEEPSLOT_TIME ? SLEEPSLOT_TIME : waiting);
+
+                Unlock();
+                // Right, lets sleep in slices of 100 ms
+                SleepMs(sleepSlot);
+                Lock();
+
+                waiting -= (waiting == Core::infinite ? 0 : sleepSlot);
+            }
+
+            uint32_t result = (((time == 0) || (IsWebSocket() == true)) ? Core::ERROR_NONE : Core::ERROR_TIMEDOUT);
+            Unlock();
+            return (result);
+        }
         uint32_t Open(const uint32_t waitTime)
         {
-            return (_channel.Open(waitTime));
+            auto start = std::chrono::high_resolution_clock::now();
```

Comment:
Agree that this is, given the waitTime, the best accurate determination of what waitTime is left, but it is a bit expensive (get times substracting 64 bits etc), thread switching with the SleepMs etc... Why not (and that might even improve performace) not do the open with a wait time of 0 (waitTime) and than only wait in the waitForLink(waitTime) ?